File: Python/Reverse String II.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        index = 0
        ns = ''
        while index < len(s):
            ns += s[index:index+k][::-1] + s[index + k:index + 2 * k]
            index += 2*k
        return ns

File: cpp/Reverse String II.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        auto start_index = s.begin();
        while (start_index < s.end()) {
            auto end_index = start_index + k > s.end() ? s.end() : start_index + k;
            reverse(start_index, end_index);
            start_index += 2 * k;
        }
        return s;
    }
};
